File: Conceitos de LPs/Subprogramas/Sobrecarga de Operadores/sobrecarga.cpp
//C++
#include <iostream>
using namespace std;
class Ponto {
public:
    int x, y;

    Ponto(int x, int y) {
	    this->x = x;
    	this->y = y;
	}

    // Sobrecarga do operador +
    Ponto operator+(Ponto& outro) {
        return Ponto(x + outro.x, y + outro.y);
    }
};

int main() {
    Ponto p1(2, 3);
    Ponto p2(4, 5);
	// Usando o operador + sobrecarregado
    Ponto p3 = p1 + p2;  

    cout << "p3 = (" << p3.x << ", " << p3.y << ")" << endl;
    return 0;
}

File: Conceitos de LPs/Subprogramas/Sobrecarga de Operadores/sobrecarga.kt
//Kotlin

// Definindo a classe Ponto
data class Ponto(val x: Int, val y: Int) {

    // Sobrecarga do operador + para somar dois pontos
    operator fun plus(outro: Ponto): Ponto {
        return Ponto(x + outro.x, y + outro.y)
    }
}

fun main() {
    // Criando dois pontos
    val p1 = Ponto(2, 3)
    val p2 = Ponto(4, 5)

    // Somando os pontos usando o operador +
    val p3 = p1 + p2

    // Exibindo o resultado
    println("Ponto resultante: $p3")
}